Tests/NimbleTests/AsynchronousTest.swift

```diff
@@ -165,6 +165,39 @@ final class AsyncTest: XCTestCase {
         }
     }
 
+    func testWaitUntilDoesNotCompleteBeforeRunLoopIsWaiting() {
+        // This verifies the fix for a race condition in which `done()` is
+        // called asynchronously on a background thread after the main thread checks
+        // for completion, but prior to `RunLoop.current.run(mode:before:)` being called.
+        // This race condition resulted in the RunLoop locking up.
+        var failed = false
+
+        let timeoutQueue = DispatchQueue(label: "Nimble.waitUntilTest.timeout", qos: .background)
+        let timer = DispatchSource.makeTimerSource(flags: .strict, queue: timeoutQueue)
+        timer.schedule(
+            deadline: DispatchTime.now() + 5,
+            repeating: .never,
+            leeway: DispatchTimeInterval.milliseconds(1)
+        )
+        timer.setEventHandler {
+            failed = true
+            fail("Timed out: Main RunLoop stalled.")
+            CFRunLoopStop(CFRunLoopGetMain())
```

Comment:
`import CoreFoundation` is required to make it compilable on Linux.

Reply:
Oh! Missed this. Yes, I'll make the change later today when I get a moment!

Reply:
@ikesyo This is done!